File: solutions/c/grains/1/grains.c
#include "grains.h"

uint64_t square(uint8_t index){
  if(index == 0) return 0;
  uint64_t res = 1;
  for(uint8_t i = 0; i < index - 1; i++){
    res = res << 1;
  }
  return res;
}

uint64_t total(void){
  uint64_t sum = 0;
  for(uint8_t i = 1; i <= 64; i++){
    sum += square(i);
  }
  return sum;
}

File: solutions/csharp/grains/1/Grains.cs
public static class Grains
{
    public static ulong Square(int n)
        => n < 1 || n > 64 ?
        throw new ArgumentOutOfRangeException() 
        : (ulong)1 << (n-1);
    

    public static ulong Total()
    {
        ulong res = 0;
        for(int i = 1; i <= 64; i++){
            res += Square(i);
        }
        return res;
    }
}